sklearn/metrics/classification.py

```diff
@@ -999,24 +981,25 @@ def precision_recall_fscore_support(y_true, y_pred, beta=1.0, labels=None,
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     present_labels = unique_labels(y_true, y_pred)
 
-    if average == 'binary' and (y_type != 'binary' or pos_label is None):
-        warnings.warn('The default `weighted` averaging is deprecated, '
-                      'and from version 0.18, use of precision, recall or '
-                      'F-score with multiclass or multilabel data or '
-                      'pos_label=None will result in an exception. '
-                      'Please set an explicit value for `average`, one of '
-                      '%s. In cross validation use, for instance, '
-                      'scoring="f1_weighted" instead of scoring="f1".'
-                      % str(average_options), DeprecationWarning, stacklevel=2)
-        average = 'weighted'
+    if average == 'binary':
+        if y_type != 'binary':
+            raise ValueError('The average is chosen as binary, but the target '
+                             'is of type %s. Please set an explicit value for '
+                             '`average`, one of %s. In cross validation use, '
+                             'for instance, scoring="f1_weighted" instead of '
+                             'scoring="f1".'
+                             % (y_type, str(average_options)))
```

Comment:
Really, what's the error out of curiosity? I couldn't find a case where `str` was needed.

Maybe there is some confusion with the usual gotcha:

```python
my_tuple = (1, 2, 3)
print('%s' % my_tuple)  # TypeError: not all arguments converted during string formatting

```

which is why it is a good habit to write this instead:
```python
my_tuple = (1, 2, 3)
print('%s' % (my_tuple,))
```

Personally I tend to use `.format`.